lib/src/intl/date_format_field.dart

```diff
@@ -670,3 +669,14 @@ class _DateFormatPatternField extends _DateFormatField {
   String padTo(int width, Object toBePrinted) =>
       parent._localizeDigits('$toBePrinted'.padLeft(width, '0'));
 }
+
+extension<T> on List<T> {
```

Comment:
Is it necessary to introduce this extension, which is only being used in one place? While I love functional interfaces, just inlining this into a for loop might be preferable, something like:
```
var results = [
  for (var i = 0; i < possibilities.length; i++)
    if (input.peek(possibilities[i].length) == possibilities[i]) i,
];
```
or even
```
var results = List.generate(possibilities.length, (i) {
  var possibility = possibilities[i];
  return input.peek(possibility.length) == possibility ? i : null;
}).whereType<int>().toList();
```

This would avoid the need to skip around the code while reading. But however you prefer :)

Reply:
Agree. If you want a helper function, just make a simple private top-level helper function. Doing it as an extension is *less* readable at the call-point.

Might be better API for a public, reusable API, but this ain't it.

Reply:
Yeah inlining is best, it will save a closure. I wrote it as an extension before I fully refactored it and realized how simple it got.

> If you want a helper function, just make a simple private top-level helper function.

I disagree. Making it an extension method or a top level method is an orthogonal concern to whether it is private (though whether it is private greatly reduces the impact of the concern).

Reply:
Just for the record then:
My take is that extension methods are *less readable* than static functions, but more *discoverable*.
Extension methods are also more verbose to declare and have a higher (non-zero) risk of conflict, when compared to a static function in the same library. It puts an extra, unnecessary, resolution step between the call site and the declaration.

Maybe I find extension methods less readable *because* we don't generally use extension methods for very specialized one-off functionality. But we don't, mostly because it's not worth it due to the other risks.

If discoverability is off the table, which it is for library local helper functions, I wouldn't use extension at all.